### PR TITLE
python 3.12, django 5.0, wagtail 6.1, postgres 12-16, psycopg 3

### PR DESCRIPTION
**Pull request by jorenham**
... and upgraded gha to use `actions/checkout@v4` and `actions/setup-python@v5`, as the cherry on top

I also took the liberty to drop support for wagtail <3, since this package isn't strictly required there.

---

*Note to self: Django 5.2 will be the first to support Python 3.13*

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -92,7 +92,7 @@ jobs:
     strategy:
       matrix:
         python: ['3.8', '3.9', '3.10']
-        django: ['3.2', '4.0']
+        django: ['3.2', '4.0', '4.1', '4.2']
```
Comment (zerolab) [inline .github/workflows/ci.yml:95]: best to drop Django 4.0 from the test matrix as all support ended 6 months ago - https://endoflife.date/django

**Comment by jorenham** [comment]
@zerolab It's been a while; would you mind taking another look? 
I've been using this fork (succesfully) in production for quite a while now.

**Review by zerolab — APPROVED** [review]
While my preference would be to bump the min requirements to 5.2 (or at least 4.1) from an LTS point of view, this looks fine.

@cnk as a big wagtail-hallo user, do you have any reservations to this going in?

**Comment by jorenham** [comment]
bump

**Comment by cnk** [comment]
@jorenham Sorry for the delay. I am working on rebasing this PR and testing on a wide range of Wagtail versions. What are you using to get icons from Wagtail 6.0 onwards? I think our team might be using a fork of wagtailfontawesome that has been updated to have recent import paths, e.g. `from wagtail.core import hooks`. Installing from https://github.com/timonweb-forks/wagtailfontawesome doesn't crash - but neither does it restore the icons in my test app.

**Comment by jorenham** [comment]
> @jorenham Sorry for the delay. I am working on rebasing this PR and testing on a wide range of Wagtail versions. 

I'm glad to see this repo is still alive!

> What are you using to get icons from Wagtail 6.0 onwards? I think our team might be using a fork of wagtailfontawesome that has been updated to have recent import paths, e.g. `from wagtail.core import hooks`. Installing from https://github.com/timonweb-forks/wagtailfontawesome doesn't crash - but neither does it restore the icons in my test app.

Eh dealer's choice I guess? I'm not really sure what it has to do with this PR, to be honest 🤷🏻

**Comment by cnk** [comment]
@jorenham if we don't do something to restore the wagtail font support or convert them all to SVGs, once I get to Wagtail 6.0 all the icons in the toolbar disappear. From your PR, this doesn't appear to be a problem for you so I was wondering what you have installed that fixes this problem.
Screenshot 2024-12-03 at 7 42 44 AM

**Comment by jorenham** [comment]
> @jorenham if we don't do something to restore the wagtail font support or convert them all to SVGs, once I get to Wagtail 6.0 all the icons in the toolbar disappear. From your PR, this doesn't appear to be a problem for you so I was wondering what you have installed that fixes this problem. Screenshot 2024-12-03 at 7 42 44 AM

I just checked, and apparently I also have no icon font installed (not sure why this went by unnoticed though) 😅

**Comment by jorenham** [comment]
> But I am reluctant to depend on something else that is end-of-life.

Given that `wagtail-hallo` is also a EOL, does it really matter (apart from the risk of a zombie apocalypse)?

**Comment by jorenham** [comment]
> Are you sure you have no icon font installed?

image

🤷🏻

**Comment by cnk** [comment]
I consolidated the commits in this PR into 2 groups and merged them in as PRs #44 and #45. Thanks @jorenham for all your hard work.

**Pull request closed without merging**